cdisc-org/cdisc-rules-engine
# Cg0014 fixes

This PR fixes issues found when trying to write CG0014

Notable changes:

1. Cache feeder now caches variable metadata grouped by dataset/domain instead of having all variables in the same dictionary. This caused the metadata for common variables to be overwritten instead of being domain specific. (e.g EPOCH is required in TA but not in LB)
2. When duplicating conditions for rules, we now operate on a copy of the rule to avoid overwriting the conditions for other concurrent runs.
3. The permissibility operation is now the VariableLibraryMetadata operation. Example usage is included in the attached test files.

Steps to test:

1. Run unit tests
2. Run the rule test command with the attached rule and dataset.
3. Verify 3 errors are shown for the AE dataset where AETERM is missing, and 1 error is shown for the LB dataset where LBTESTCD is missing.
[cg0014-test-data.zip](https://github.com/cdisc-org/cdisc-rules-engine/files/11851378/cg0014-test-data.zip)
